extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java

```diff
@@ -0,0 +1,85 @@
+package io.quarkus.smallrye.graphql.runtime;
+
+import java.io.StringReader;
+
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonReaderFactory;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.security.identity.CurrentIdentityAssociation;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.smallrye.graphql.execution.ExecutionService;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Handler that does the execution of GraphQL Requests
+ */
+public abstract class SmallRyeGraphQLAbstractHandler implements Handler<RoutingContext> {
+
+    private final CurrentIdentityAssociation currentIdentityAssociation;
+    private final CurrentVertxRequest currentVertxRequest;
+
+    private volatile ExecutionService executionService;
+
+    private static final JsonBuilderFactory jsonObjectFactory = Json.createBuilderFactory(null);
```

Comment:
Any reason to not use Jackson (which is the recommended JSON support in Quarkus)?

Reply:
SmallRye GraphQL use JsonP and JsonB (JsonB support from the Spec, so fields can be transformed with JsonB), so the method being called here (in the `ExecutionService` in SmallRye) expects a JsonObject from JsonP. We can look into supporting Jackson in SmallRye GraphQL, but we already have (anyway because of JsonB) JsonP available.